vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt

```diff
@@ -36,74 +38,79 @@ import java.util.Objects
             throw ProofTypeNotSupportedException("Proof Type available in received credentials is not matching with supported proof terms")
         }
         return try {
             val canonicalizer = URDNA2015Canonicalizer()
             val canonicalHashBytes: ByteArray = canonicalizer.canonicalize(ldProofWithJWS, vcJsonLdObject)
-            Log.i(tag,"Completed Canonicalization for the received credentials.")
             val signJWS: String = ldProofWithJWS.jws
             val jwsObject: JWSObject = JWSObject.parse(signJWS)
             val vcSignBytes: ByteArray = jwsObject.signature.decode()
             val publicKeyJsonUri: URI = ldProofWithJWS.verificationMethod
             val publicKeyObj = getPublicKeyFromVerificationMethod(publicKeyJsonUri)
             if (Objects.isNull(publicKeyObj)) {
-                Log.e(tag,"Public key object is null, returning false")
                 throw PublicKeyNotFoundException("Public key object is null")
             }
-            Log.i(tag,"Completed downloading public key from the issuer domain and constructed public key object.")
             val actualData: ByteArray = JWSUtil.getJwsSigningInput(jwsObject.header, canonicalHashBytes)
             val jwsHeader: String = jwsObject.header.algorithm.name
-            Log.i(tag,"Performing signature verification after downloading the public key.")
             verifyCredentialSignature(jwsHeader, publicKeyObj, actualData, vcSignBytes)
         } catch (e: Exception) {
             when (e) {
                 is PublicKeyNotFoundException,
                 is SignatureVerificationException -> throw e
                 else -> {
-                    Log.e(tag, "Error while doing verification of verifiable credential", e)
                     throw UnknownException("Error while doing verification of verifiable credential")
                 }
             }
         }
     }
 
+    @Throws(CertificateException::class, KeyStoreException::class, KeyManagementException::class)
     private fun getPublicKeyFromVerificationMethod(publicKeyJsonUri: URI): PublicKey? {
-        try {
-            val restTemplate = RestTemplate()
-            val response: ObjectNode? =
-                restTemplate.exchange(publicKeyJsonUri, HttpMethod.GET, null, ObjectNode::class.java).body
-            val publicKeyPem: String? = response?.get(CredentialVerifierConstants.PUBLIC_KEY_PEM)?.asText()
-            Log.i(tag,"public key download completed.")
-            val strReader = StringReader(publicKeyPem)
-            val pemReader = PemReader(strReader)
-            val pemObject: PemObject = pemReader.readPemObject()
-            val pubKeyBytes: ByteArray = pemObject.content
-            val pubKeySpec = X509EncodedKeySpec(pubKeyBytes)
-            val keyFactory = KeyFactory.getInstance("RSA")
-            return keyFactory.generatePublic(pubKeySpec)
+        return try {
+            val okHttpClient = OkHttpClient.Builder().build().newBuilder().build()
```

Comment:
@sree96 , last time when we tried to integrate a library with Inji, OkHttpClient was not compatible android JVM. can we try to integrate with inji app and verify once. or we can use other HTTP Clients

Reply:
We have tried this @vijay151096 , The older implementation which was java.net.http.HttpClient was the one incompatible with Android. These changes with OkHttpClient we have already tested on Android and is working fine. 

cc: @Alka1703

Reply:
Okay Fine then, all good here.